src/main/scala/uk/gov/nationalarchives/draftmetadatavalidator/Lambda.scala

```diff
@@ -42,73 +43,98 @@ class Lambda extends RequestHandler[java.util.Map[String, Object], APIGatewayPro
+        unexpectedFailureResponse.withBody(s"Unexpected validation problem:${error.getMessage}")
+        IO(unexpectedFailureResponse)
+      })
+      .unsafeRunSync()(cats.effect.unsafe.implicits.global)
+  }
+
+  private def doValidation(validationParameters: ValidationParameters): IO[ErrorFileData] = {
+    val s3Files = S3Files(S3Utils(s3Async(s3Endpoint)))
+    val validationProgram = for {
+      _ <- s3Files.downloadFile(bucket, validationParameters)
+      _ <- validUTF8(validationParameters)
+      csvData <- loadCSV(validationParameters)
+      _ <- validateRequired(csvData, validationParameters)
+      _ <- validateMetadata(validationParameters, csvData)
+    } yield ErrorFileData(validationParameters, FileError.None, List.empty[ValidationErrors])
+
+    // all validations will raise ValidationExecutionError if they do not pass
+    validationProgram.handleError({
+      case validationExecutionError: ValidationExecutionError => validationExecutionError.errorFileData
+      case err =>
+        logger.error(s"Error doing validation for consignment:${validationParameters.consignmentId.toString} :${err.getMessage}")
+        val singleError = Error("Validation", validationParameters.consignmentId.toString, "UNKNOWN", err.getMessage)
+        val validationErrors = ValidationErrors(validationParameters.consignmentId.toString, Set(singleError))
+        ErrorFileData(validationParameters, FileError.UNKNOWN, List(validationErrors))
+    })
   }
 
   // use a required schema, pass one row of data that will return missing required fields, change row identifier
   // to consignmentID. Can then use to help populate required on UI
   // just hack code
-  private def validateRequired(csvData: List[FileRow], consignmentID: String): Either[Unit, ErrorFileData] = {
+  private def validateRequired(csvData: List[FileRow], validationParameters: ValidationParameters): IO[Unit] = {
     // TODO: To be implemented TDRD-62
-    Left()
+    IO.unit
   }
 
-  // Check file is UTF
-  // maybe have new ValidationProcess.FILE_CHECK that can be used file integrity checks
-  private def validUTF(draftMetadata: DraftMetadata): IO[Either[Unit, ErrorFileData]] = {
-    // TODO: To be implemented TDRD-61
-    IO(Left())
-  }
+  private def validUTF8(validationParameters: ValidationParameters): IO[Unit] = {
+    val filePath = getFilePath(validationParameters)
+
+    def utf8FileErrorData = {
+      val messageKey = "FILE_CHECK.UTF.INVALID"
+      val message = if (messageProperties.containsKey(messageKey)) messageProperties.getProperty(messageKey) else messageKey
+      val singleError = Error("FILE_CHECK", validationParameters.consignmentId.toString, "UTF8", message)
+      val validationErrors = ValidationErrors(validationParameters.consignmentId.toString, Set(singleError))
+      ErrorFileData(validationParameters, FileError.UTF_8, List(validationErrors))
+    }
+
+    def checkBOM(inputStream: FileInputStream) = {
```

Comment:
Feel free to ignore but instead of string manipulation we can do something like this?
```
    def checkBOM(inputStream: FileInputStream): IO[Unit] = {
      Resource.fromAutoCloseable(IO(inputStream)).use { inputStream =>
        val bom = Array[Byte](0xEF.toByte, 0xBB.toByte, 0xBF.toByte)
        val firstBytes = new Array[Byte](3)
        if (inputStream.read(firstBytes) == 3 && firstBytes.sameElements(bom)) {
          IO.unit
        } else {
          IO.raiseError(ValidationExecutionError(utf8FileErrorData, List.empty[FileRow]))
        }
      }
    }
```

Reply:
Much nicer